runtime/src/main/cpp/utf8/core.h

```diff
@@ -251,23 +271,14 @@ namespace internal
         }
 
         if (err == UTF8_OK) {
-            // Decoding succeeded. Now, security checks...
-            if (utf8::internal::is_code_point_valid(cp)) {
```

Comment:
These check moved to `get_sequence_N` functions to detect invalid sequence earlier.